### DISCUSSION TITLE
Broadcasting messages to Channel | WPPConnect -> Newsletter Layer

### DISCUSSION THREAD

**Discussion by cseguinlz** [3mora2/WPP_Whatsapp#31] [Q&A] [answered]
Hi team, 
I'm trying to push messages to one Whatsapp channel of mine. Got a couple of questions see if you can help:

- Does newsletter = channel? from [WPPConnect](https://github.com/wppconnect-team/wppconnect). I suspect it does, just looking for some confirmation, and if so...
- Would you be migrating the Newsletter Layer  at some point?

I know I should go ahead and just do it myself, but seriously, don't feel qualified, I've been playing with python for just one week now.

Thanks team and great effort, really really helpful.

**Comment by 3mora2** [comment]
Yes, newsletter same channel
Yes, will migrate but wait until wppconnect finish all method
To send message to channel get id of it and use method like sendText

**Reply by cseguinlz** [reply]
So I did but it comes back with:

```
playwright._impl._errors.Error: Error: wid error: invalid wid
    at e (https://web.whatsapp.com/:2:4914)
    at new f (https://web.whatsapp.com/app.036ba334d9778b01397b.js:306:222958)
    at Object.c [as createWid] 
    
```
Noticed that def `valid_chatId(chatId)` checks for `@newsletter` suffix so I added it to the id, but same result.

sendText() to a phone number works fine.  Maybe I'm not getting the right Channel ID, just using the end segment of channels url https://www.whatsapp.com/channel/<using this bit as ID>

**Reply by 3mora2** [reply, marked as answer]
you can send like chat get channel id and send
`client.sendText("************@newsletter", message)`
![-5918217552817931178_121](https://github.com/3mora2/WPP_Whatsapp/assets/66757189/21a7e61e-3023-4f59-806a-554288fe46e9)

or open it and run in console 
`document.querySelector("[data-id*='@newsletter']").getAttribute("data-id").split("_")`

Answer from https://t.me/WPP_Whatsapp

**Reply by cseguinlz** [reply]
oh man... thank you so much, you've made my week. 
I suspected the id wasn't right and it made sense for it to be the one in the url. What a relief, thanks a lot

شكراً جزيلاً لك
